packages/integration-karma/test/template/attribute-hidden/x/test/test.html

```diff
@@ -0,0 +1,19 @@
+<template>
+    <!-- hidden="true" or hidden="false" is a compilation error -->
+
+    <p class="test-case">should be false when missing</p>
```

Comment:
I was under the original impression that for standard elements, all the attributes from the template remain as attributes and the template compiler converts the attributes to props for custom elements only. But looking closer at the compiler code, the boolean attributes on standard elements are also converted to props. Does anyone recall what is the reasoning behind this?

Reply:
I don't know the reasoning, it's being treated as such for a while now (https://github.com/salesforce/lwc/pull/172/files#diff-866c3ea9738d7f68e8291a9ecbcbebc1R203).

Something that comes to my mind ... what do you think should be the behavior here?
```html
<template>
    <p hidden={computed}>This is a hidden text</p>
</template>
```
```js
import { LightningElement } from 'lwc';

export default class App extends LightningElement {
    computed = false;
}
```
 Notice that `elm.setAttribute("hidden", false)`, will actually hide the element.

Reply:
It seems to me that for standard elements, anything that is an expression should be set as a prop and anything that is not an expression should be set as an attribute.

> Notice that elm.setAttribute("hidden", false), will actually hide the element.

I guess this is expected since `setAttribute()` will just cast the value into a string before setting it on the element.

Reply:
> anything that is an expression should be set as a prop and anything that is not an expression should be set as an attribute.

I don't think we should have a different code path depending if the associated value is an expression or not. Some properties reflect attributes but not the opposite. For example if you want to set a `foo` attribute on a random HTML element, it can only be done via an attribute. IMHO, we should strive to make a simple rule to disambiguate this.

What do you think about some like, _All the attributes declared in the template on standard HTML elements are set via attributes. Know boolean attributes (eg. `hidden`, `required`) are set if their value is not is truthy or otherwise, the attribute is omitted. For the rest of the attribute (eg. `class`, `data-*`, `foo-bar`), the values are set if they are not nullish._

Reply:
> All the attributes declared in the template on standard HTML elements are set via attributes. Know boolean attributes (eg. hidden, required) are set if their value is not is truthy or otherwise, the attribute is omitted. For the rest of the attribute (eg. class, data-*, foo-bar), the values are set if they are not nullish.

I think I agree with this, although the second sentence is a little ambiguous. Did you mean to say that known boolean attributes are added only if the value is truthy? And anything else is only set if the value is neither `undefined` nor `null`?

I think we would also need to have a special case for attributes that are only used to initialize values.

Reply:
> Did you mean to say that known boolean attributes are added only if the value is truthy? And anything else is only set if the value is neither undefined nor null?
Yes and yes.

> I think we would also need to have a special case for attributes that are only used to initialize values.
Good point. Do you have anything else in mind than the `checked` and the `value` properties on the `HTMLInputElement`?

Reply:
the reason is the removal... we don't have a good signal for the diffing to remove the value or an attribute.

Reply:
@caridy I don't think the removal is an issue anymore with the proposal: https://github.com/salesforce/lwc/pull/1952#discussion_r448822745.